File: String Encode and Decode/python/stringEncodeAndDecode.py
from typing import List


def encode(strs: List[str]) -> str:
    res = ""
    for s in strs:
        res += str(len(s)) + "#" + s
    
    return res


def decode(s: str) -> List[str]:
    res = []
    i, j = 0, 0

    while i < len(s):
        while s[j] != '#':
            j += 1
        length = int(s[i:j])
        i = j + 1
        j = i + length
        res.append(s[i:j])
        i = j

    return res

strs = ["neet","code","love","you"]
print(decode(encode(strs)))

File: String Encode and Decode/ts/stringEncodeAndDecode.ts
function encode(strs: string[]): string {
    let res = ""
    for (const s of strs)
        res += `${s.length}#${s}`

    return res
}

function decode(s: string): string[] {
    const res: string[] = []
    let i = 0
    let j = 0

    while (i < s.length) {
        while (s[j] != "#") j++;
        const len = parseInt(s.substring(i, j))
        i = j + 1
        j = i + len
        res.push(s.substring(i, j))
        i = j
    }

    return res
}

const strs = ["neet","code","love","you"]
console.log(decode(encode(strs)))